### ISSUE TITLE
Expression working only on historical data

### ISSUE THREAD

**Issue by vitorboschi** [bosun-monitor/bosun#806]
I'm doing some tests on Bosun, but I can't make the following expression to work:

q("max:1m-max:ping.timeout{dst_host=reader*}", "1h", "")

The generated query is correct:
end=2015/03/19-17:33:11&m=max:1m-max:ping.timeout{dst_host=*}&start=2015/03/19-16:33:11

If I click the query link, the results are displayed correctly (and also if I specify a past date on the expression page). It just doesn't show any data on the expression page if I don't set any date/time

Is it a problem if most of the data currently in openTSDB was not sent through Bosun, but directly to openTSDB?

I'm on rev 236e300ee8a5fdd601e7ffc1768f47adff2d80d8

**Comment by kylebrandt** [comment]
It maters if you use a wildcard to in the tag value.  OpenTSDB only
supports an asterisk as the only character (so it is all or nothing).
Expanding the glob is an extension bosun does, and it learns about possible
expansions based on the data relayed through it.

**Comment by vitorboschi** [comment]
I don't think the problem is in the expansion, since it already seens to know possible tag values, because I can execute queries using * on it. Is there anything I can test, or some way to enable more debug info?

**Comment by kylebrandt** [comment]
Hit alt-p tp bing up mini profiler, then query. There should be stuff in
the lower left and I believe it will show the opentsdb query issued.

**Comment by vitorboschi** [comment]
I couldn't find the openTSDB query in the mini profiler, just the expression inside Bosun itself. The query generated seens correct, because it's shown by bosun as a link, and it indeed works as expected if I click it.
For some reason, it's just not querying openTSDB, or not processing its response

**Comment by vitorboschi** [comment]
Profiler screenshot: 
![bosun](https://cloud.githubusercontent.com/assets/71193/6738106/93f0a0c8-ce4e-11e4-9cda-cb4df8efa63e.png)

**Comment by kylebrandt** [comment]
Now that you have only a  `*` it should work. My next guess would be that the time on the Bosun host doesn't match the time of OpenTSDB.

**Comment by vitorboschi** [comment]
Time shouldn't be a problem, since both Bosun and OpenTSDB are on the same machine. I'm trying to add some debug info in the code. 
I can see that func Query in expr/funcs.goexpr/funcs.go is executing without errors, So I'm trying to go add some debug to the function which queries OpenTSDB. I've tried

func (r *Request) Query(host string) (ResponseSet, error) { 

in tsdb.go, but it doesn't seem to be called (even when I set the time manually and the expression works as expected). I guess it's the wrong function.

Any tips on how to proceed?
